spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestComputePartitionStatsAction.java

```diff
@@ -0,0 +1,291 @@
+    // foo, B -> 2 records,
+    // bar, A -> 2 records,
+    // bar, B -> 1 record
+    sql(
+        "INSERT into %s values (0, 'foo', 'A'), (1, 'foo', 'A'), (2, 'foo', 'B'), (3, 'foo', 'B')",
+        tableName);
+    Table table = validationCatalog.loadTable(tableIdent);
+    Snapshot snapshot1 = table.currentSnapshot();
+    sql("INSERT into %s values(4, 'bar', 'A'), (5, 'bar', 'A'), (6, 'bar', 'B')", tableName);
+    table.refresh();
+    Snapshot snapshot2 = table.currentSnapshot();
+    sql("INSERT into %s values(7, 'foo', 'A'), (8, 'foo', 'A')", tableName);
+    // snapshot3 is unused for partition stats as the same partition is modified by snapshot4
+
+    // delete one record of foo, A
+    sql("DELETE FROM " + tableName + " WHERE c1=1");
+    table.refresh();
+    Snapshot snapshot4 = table.currentSnapshot();
+
+    assertThat(table.partitionStatisticsFiles()).isEmpty();
+
+    SparkActions actions = SparkActions.get();
+    PartitionStatisticsFile statisticsFile =
+        actions.computePartitionStats(table).execute().statisticsFile();
+    assertThat(statisticsFile.fileSizeInBytes()).isGreaterThan(0);
+    assertThat(statisticsFile.snapshotId()).isEqualTo(snapshot4.snapshotId());
+    // check table metadata registration
+    assertThat(table.partitionStatisticsFiles()).containsExactly(statisticsFile);
+
+    Types.StructType partitionType = Partitioning.partitionType(table);
+    Schema dataSchema = PartitionStatsHandler.schema(partitionType);
+    validatePartitionStats(
+        statisticsFile,
+        dataSchema,
+        Tuple.tuple(
+            partitionRecord(partitionType, "foo", "A"), // partition
+            0, // specId
+            4L, // dataRecordCount (total 4 records for this partition)
+            2, // dataFileCount
+            datafileSize("foo", "A"), // totalDataFileSizeInBytes
+            1L, // positionDeleteRecordCount (from delete operation)
+            1, // positionDeleteFileCount (from delete operation)
+            0L, // equalityDeleteRecordCount
+            0, // equalityDeleteFileCount
+            null, // totalRecords
+            snapshot4.timestampMillis(), // lastUpdatedAt (last modified by snapshot4)
+            snapshot4.snapshotId() // lastUpdatedSnapshotId
+            ),
+        Tuple.tuple(
+            partitionRecord(partitionType, "foo", "B"), // partition
+            0, // specId
+            2L, // dataRecordCount
+            1, // dataFileCount
+            datafileSize("foo", "B"), // totalDataFileSizeInBytes
+            0L, // positionDeleteRecordCount
+            0, // positionDeleteFileCount
+            0L, // equalityDeleteRecordCount
+            0, // equalityDeleteFileCount
+            null, // totalRecords
+            snapshot1.timestampMillis(), // lastUpdatedAt (added by snapshot1)
+            snapshot1.snapshotId() // lastUpdatedSnapshotId
+            ),
+        Tuple.tuple(
+            partitionRecord(partitionType, "bar", "A"), // partition
```

Comment:
it's obvious from the parameter that this is the partition, so no need to have a comment at the end